dagify/converter/utils.py

```diff
@@ -200,13 +236,117 @@ def format_table_data(title, columns, rows):
     return table_data
 
 
-def generate_json(statistics, table_data, output_file_path):
+def generate_json(statistics, job_table_data, schedule_table_data,output_file_path):
     """Creates a JSON file with intro text, table data, and conclusion text"""
 
     data = {
         "High_Level_Info": statistics,
-        "table_data": table_data,
+        "Job_info_table": job_table_data,
+        "Schedule_info_table": schedule_table_data
     }
     json_file_path = f"{output_file_path}/report.json"
     with open(json_file_path, "w") as json_file:
         json.dump(data, json_file, indent=2)  # indent for better readability
+
+
+def get_job_info(file_path):
+    """Function to get and return a dictionary of job_name and its task_type"""
+    if not file_path.endswith('.xml'):
+        raise ValueError(f"Invalid file format: {file_path}. Only XML files are supported.")
+    job_info_list = []
+    tree = ET.parse(file_path)
+    root = tree.getroot()
+    # Find all JOB elements
+    for job in root.findall('.//JOB'):  # Find all JOB elements
+        job_name = job.attrib['JOBNAME']
+        task_type = job.attrib['TASKTYPE'].lower()
+        job_info_list.append({'job_name': job_name, 'task_type': task_type})
+    return job_info_list
+
+
+def get_job_statistics(job_info_list, config_task_type):
+    """Function to calculate and return job_name statistics"""
+    unconverted_job_name = [job['job_name'] for job in job_info_list
+                            if job['task_type'] not in config_task_type]
+    converted_job_name = [job['job_name'] for job in job_info_list
+                          if job['task_type'] in config_task_type]
+    conv_job_count = len(converted_job_name)
+    non_converted_job_percent, converted_job_percent = \
+        calculate_percentages(unconverted_job_name, converted_job_name)
+    return unconverted_job_name, converted_job_name, \
+        non_converted_job_percent, converted_job_percent,conv_job_count
+
+
+def get_template_name(self, job_type):
+    """Function to return template_name for a particular job_type"""
+    for mapping in self.config["config"]["mappings"]:
+        if mapping["job_type"] == job_type.upper():
+            return mapping["template_name"]
+    # no match found
+    return None
+
+
+def calculate_cron_schedule(task):
```

Comment:
same here, we should avoid copy pasting code but rather importing utility functions if they are being used somewhere else.